Add mechanism to automatically switch author and committer between commits

sykesm opened the pull request:
A while ago, my team made the switch to `git-duet` but some of the folks went back to `git-pair` since it randomly selects an author from the pairs; they like this as it helps distribute the 'green dots' between people while they're pairing.

Since I'm not a fan of how `git-pair` generates fictitious email addresses or loses track of one of the authors, I'd like to get everyone back to using pair.

This pull request adds code to swap the author and committer at the end of every commit when the `GIT_DUET_ROTATE_AUTHOR` environment variable is set. This mode of operation should distribute the 'green dots' in fair and deterministic way.

jszwedko commented on src/go-git-duet/git_config.go:
```diff
@@ -24,18 +25,21 @@ func (gc *GitConfig) ClearCommitter() (err error) {
 		return err
 	}
-	if err = gc.unsetKey("git-committer-name"); err != nil {
+	if err = gc.unsetKey("git-committer-email"); err != nil {
```
Oops! Thanks for catching this copy/paste error.

jszwedko commented:
@sykesm I like this idea, thanks for putting this together! Overall I'm very :+1:, but I am curious about the addition of `getGitconfig`. Previously I was just relying on the `git config` to read from local and fall back to global when necessary, is there something I'm missing that required this addition? Or why it is only necessary for `git duet-commit`?

jszwedko commented:
Aaah I see #11 now. Let me think about that for a moment.

jszwedko commented:
@sykesm I put together #12 which I think feels a little better to address #11, what do you think?

sykesm commented:
@jszwedko I think #12 addresses the stale committer name from #11 but I don't think it addresses the second issue of display vs actual that's due to forcing the global config in `duet` but relying the default search order in `duet-commit`.

I did throw https://github.com/git-duet/git-duet/commit/6f175494dc363ee184642f864e5effbd0c9f6bc0 together yesterday which, in addition to explicitly getting the config source, delays forcing the --global flag until the configuration needs to be updated. You can see that it effected `duet`, `solo`, and `duet-commit`.

If you're still unsure about determining the config source up front, we could probably combine the two.

Thanks.

jszwedko commented:
@sykesm Aaah, I see what you are saying now. I like the clarity of what you have here to solve that.

The only other question I have is: what do you think about just treating `Global: false` as `Local: true` rather than introducing a new field since the distinction is binary? I worry that the extra field will somehow cause us to have a struct that has `Global: true` and `Local: true` unintentionally at some point.

Other than that I like what you have in 6f175494dc363ee184642f864e5effbd0c9f6bc0 and would like to merge that in at the same time as this PR.

sykesm commented:
@jszwedko heh. Yeah, I thought about that. The problem I ran into is that if we always treat lack of `--global` as `--local`, we'll have to manually search in more cases. I'm not sure that's a problem.

Let me see if I can swizzle what's there to improve it a bit and get your opinion. Got a few balls in the air right now so it might not be until tomorrow.

Thanks.

jszwedko commented:
Yeah, I see what you are saying. Maybe just using an "enum" would be sufficient then, like:

``` go
type scope int

const (
    Default scope = iota
    Local
    Global
)
```

That would at least limit the invalid state of `Local: true` and `Global: true`. I'll think about this too.

jszwedko commented:
This is what it'd look like as a faux-enum: 3982ccf04913b7616dfeeb662aded270b1821941 using your commit as parent (also used `GetAuthorConfig` in `git-solo` just for parity even though it isn't strictly necessary).

sykesm commented:
@jszwedko, I appreciate your patience. Hopefully this addresses the remaining concerns. Please let me know if there's anything else you're looking for.

Thanks.

jszwedko commented:
@sykesm no worries, thanks for the contribution! I think I'm satisfied with this, will rebase into master.

jszwedko commented:
Rebased the commits onto master:
5c3b9f6
323ecb9
5e40d10

sykesm commented:
@jszwedko thanks. Two questions.
1. Would you be willing to accept a PR to honor the 'global: true' option in the .git-authors file as is done with [`git-pair`](https://github.com/pivotal/git_scripts#git-pair)?
2. Are you planning on cutting a release with this support and making it available via the homebrew tap?

Thanks.

jszwedko commented:
1. Yep! I definitely wouldn't mind seeing all of what we currently allow to be configured via the environment to also be configurable via the YML file.
2. Also yes, trying to get Travis to cut a new release now.

jszwedko commented:
Released 0.2.0 and updated tap
